solr/core/src/java/org/apache/solr/search/facet/UniqueAgg.java

```diff
@@ -85,7 +85,7 @@ public void merge(Object facetResult, Context mcontext) {
       List<?> vals = (List<?>) map.get(VALS);
       if (vals != null) {
         if (values == null) {
-          values = new HashSet<>(vals.size() * 4);
+          values = CollectionUtil.newHashSet(vals.size() * 4);
```

Comment:
I agree that some of the other * some multiple seem suspicious. I'll take a look. at least in UniqueAgg, I don't know what the size should be at all. Its not clear :/ Its been *4 since 2015 when it was added so going to leave it for now.